### PR TITLE
openai[patch]: pass message name

### PR DESCRIPTION
**Pull request by efriis**

### CODE REVIEW

File: libs/partners/openai/pyproject.toml [hunk 1]
```diff
@@ -12,7 +12,7 @@ license = "MIT"
 
 [tool.poetry.dependencies]
 python = ">=3.8.1,<4.0"
-langchain-core = "^0.1.16"
+langchain-core = "^0.1.25"
```
Comment (efriis) [inline libs/partners/openai/pyproject.toml:15]: had to bump core for basemessage.name

File: libs/partners/openai/langchain_openai/chat_models/base.py [hunk 2]
```diff
@@ -135,19 +144,15 @@ def _convert_message_to_dict(message: BaseMessage) -> dict:
...
-        message_dict = {
-            "role": "tool",
-            "content": message.content,
-            "tool_call_id": message.tool_call_id,
-        }
+        message_dict["role"] = "tool"
+        message_dict["tool_call_id"] = message.tool_call_id
+
+        # tool message doesn't have name: https://platform.openai.com/docs/api-reference/chat/create#chat-create-messages
+        del message_dict["name"]
```
Comment (jacoblee93) [inline libs/partners/openai/langchain_openai/chat_models/base.py:155]: Should we log a warning here? Will OpenAI reject this?

File: libs/partners/openai/langchain_openai/chat_models/base.py [hunk 3]
```diff
@@ -79,8 +79,9 @@ def _convert_dict_to_message(_dict: Mapping[str, Any]) -> BaseMessage:
         The LangChain message.
     """
     role = _dict.get("role")
+    name = _dict.get("name")
```
Comment (jacoblee93) [inline libs/partners/openai/langchain_openai/chat_models/base.py:82]: This doesn't appear to be in the TS SDK types

File: libs/partners/openai/langchain_openai/chat_models/base.py (hunk 2 above)
Comment (efriis) [reply libs/partners/openai/langchain_openai/chat_models/base.py:170]: OpenAI rejects. Think I should raise an error?

Comment (efriis) [reply libs/partners/openai/langchain_openai/chat_models/base.py]: changed to pass through unless it's none

**Pull request merged**